rewriter/SourceRewriter.cpp

```diff
@@ -509,8 +528,13 @@ class FnPtrExpr : public RefactoringCallback {
                        hasEitherOperand(expr(
                            ignoringImplicit(equalsBoundNode("fnPtrExpr"))))));
 
+    auto param_expr = hasParent(implicitCastExpr(hasParent(callExpr(
+        forEachArgumentWithParam(equalsBoundNode("fnPtrExpr"),
+                                 parmVarDecl().bind("fnPtrParamDecl"))))));
+
     StatementMatcher fn_ptr_expr =
-        expr(fn_expr, unless(call_expr), unless(in_bin_op));
+        expr(fn_expr, unless(call_expr), unless(in_bin_op),
+             anyOf(param_expr, anything()));
```

Comment:
Why `anyOf(anything())`?

Reply:
Because `expr()` is implicitly `allOf` and we want to match things that aren't call arguments. Is there a better way to do this? I basically just wanted to bind the param decl to a name with the `param_expr` matcher. Probably deserves a comment at least though.

Reply:
I vaguely remember running into a similar issue with the callsite AST matcher but I'll take a closer look this afternoon.

Reply:
It's not quite the same thing I ran into before, but I think using `optionally` might make your intent more clear.
```suggestion
             optionally(param_expr));
```

Reply:
I missed that operator, that's perfect, thanks.